examples/scripts/checkfiles.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+# The script compares files with the same name from two different directories.
+# The names of the files are defined in the scripts, the names of the two different
+# directories are passed as parameters when executing the script.
+# The listed files are produced when executing the test suite of FairRoot
+# The script immediately fails if a difference between files is found
+# The actaul comparison is done using a ROOT macro which comapares the leaves of
+# the ROOT tree ony by one.
+
+if [[ "$#" -ne 2 ]]; then
+  echo "Wrong number of arguments"
+  echo "The script expects two directories as arguments" 
+  exit 1 
+fi
+
+dir1=$1
+dir2=$2
+
+files=( 
+simulation/Tutorial1/macros/tutorial1_TGeant4_pions.mc_p2.000_t0_n1.root
+simulation/Tutorial1/macros/tutorial1_TGeant4_pions.mc_p2.000_t0_n10.root
+simulation/Tutorial1/macros/tutorial1_pythia8_TGeant4_pions.mc_p2.000_t0_n10.root
+simulation/Tutorial1/macros/tutorial1_pythia6_TGeant4_pions.mc_p2.000_t0_n10.root
+simulation/Tutorial1/macros/tutorial1_urqmd_TGeant4.mc.root
+
+simulation/Tutorial2/macros/tutorial2_pions.mc_p2.000_t0_n10.root
+simulation/Tutorial2/macros/tutorial2_pions.mc_p2.000_t0_n10.sg1.root
+simulation/Tutorial2/macros/tutorial2_pions.mc_p2.000_t0_n20.sg2.root
+simulation/Tutorial2/macros/tutorial2_pions.mc_p2.000_t0_n130.bg.root
+#simulation/Tutorial2/macros/digis.mc.root
+#simulation/Tutorial2/macros/digis.mix.mc.root
+
+simulation/Tutorial4/macros/data/testrun_align_TGeant4.root
+simulation/Tutorial4/macros/data/testreco_align_TGeant4.root
+#simulation/Tutorial4/macros/data/test.ana.root
+
+simulation/rutherford/macros/data/test_TGeant4.mc.root
+simulation/rutherford/macros/data/test1_TGeant4.mc.root
+
+advanced/propagator/macros/prop.mc.root
+#advanced/propagator/macros/prop.rk.cal.root
+
+advanced/Tutorial3/macro/data/testrun_TGeant4.root
+advanced/Tutorial3/macro/data/testdigi_TGeant4.root
+advanced/Tutorial3/macro/data/testreco_TGeant4.root
+advanced/Tutorial3/macro/data/testDiRePr_TGeant4.root
+#advanced/Tutorial3/macro/data/testrecotimebased_TGeant4.root
+
+#MQ/serialization/data_io/testinput1.root
+#MQ/serialization/data_io/outputEx1Test.root
+#MQ/serialization/data_io/testinput2.root
+#MQ/serialization/data_io/outputEx2Test.root
+#MQ/pixelDetector/macros/pixel_TGeant4.mc.root
+#MQ/pixelDetector/macros/pixel_TGeant4.digi.root
+#MQ/pixelDetector/macros/pixel_TGeant4.digiToBin.root
+#MQ/pixelDetector/macros/MQ.pixel_TGeant4.hits.root
+#MQ/pixelSimSplit/run/MQ.simulation_TGeant4.data.root
```

Comment:
We should generate this list with CMake, having it hardcoded like this will be a nightmare to maintain.

Reply:
I agree but I am not sure if this will be so easy. For the time being it would be good if we could keep the hard coded list.

Reply:
An idea would be to glob for `${dir1}/**/*.root` and then iterate this list of files, if such a root file also exists in `$dir2`. We may have to add then another sanity check to the `TreeCompareAuto.C` macro that it will skip non-FairRoot root files, if there were any.

Reply:
The problem is that not all ROOT files should be tested. Not all off them contain the tree. That means that we have to filter the list afterwards to get only the list of files we want. In my opinion this isn't better than having a fixed list.

One solution would be to provide some benchmark files you want to compare and loop over these files and compare them with the newly produced ones. The downside of this is that we have to provide the benchmark files somehow.

Reply:
Another approach would be:

* Add a CMake list variable `FAIRROOT_SIMTREE_FILES_FOR_TESTING`
* Append file names to it in each of the local `CMakeLists.txt` files from each example/test the file belongs to
* Then `configure_file()` this `checkfiles.sh` script in a way that the contents of the CMake variable is added

Then the "authority" of naming relevant root files stays local to the individual example/test/lib. What do you think?